source/npm.d.ts

```diff
@@ -0,0 +1,493 @@
+		*/
+		restart?: string;
+
+		/**
+		Run with the `npm restart` command, after `restart`. Note: `npm restart` will run the `stop` and `start` scripts if no `restart` script is provided.
+		*/
+		postrestart?: string;
+	} & {
+		[scriptName: string]: string;
+	}
+
+	/**
+	Names of package dependencies that are bundled when the package is publihed.
+	*/
+	export type BundledDependency = string[];
+
+	/**
+	Dependencies of the package. The version range is a string which has one or more space-separated descriptors. Dependencies can also be identified with a tarball or Git URL.
+	*/
+	export interface Dependency {
+		[packageName: string]: string;
+	}
+
+	export interface TypeScriptConfiguration {
+		/**
+		Location of the bundled TypeScript declaration file.
+		*/
+		types?: string;
+
+		/**
+		Location of the bundled TypeScript declaration file. Alias of `types`.
+		*/
+		typings?: string;
+	}
+
+	export interface YarnConfiguration {
+		/**
+		Selective version resolutions. Allows the definition of custom package versions inside dependencies without manual edits in the `yarn.lock` file.
+		*/
+		resolutions?: Dependency;
+	}
+
+	export interface JSPMConfiguration {
+		/**
+		JSPM configuration.
+		*/
+		jspm?: PackageJson;
+	}
+}
+
+export type PackageJson = {
+	/**
+	The name of the package.
+	*/
+	name?: string;
+
+	/**
+	Package version, parseable by [`node-semver`](https://github.com/npm/node-semver).
+	*/
+	version?: string;
+
+	/**
+	Package description, listed in `npm search`.
+	*/
+	description?: string;
+
+	/**
+	Keywords associated with package, listed in `npm search`.
+	*/
+	keywords?: string[];
+
+	/**
+	The URL to the package's homepage.
+	*/
+	homepage?: LiteralUnion<'.', string>;
+
+	/**
+	The URL to the package's issue tracker and/or the email address to which issues should be reported.
+	*/
+	bugs?: PackageJson.BugsLocation;
+
+	/**
+	The license for the package.
+	*/
+	license?: string;
+
+	/**
+	The licenses for the package.
+	*/
+	licenses?: {
+		type?: string;
+		url?: string;
+	}[];
+
+	author?: PackageJson.Person;
+
+	/**
+	A list of people who contributed to the package.
+	*/
+	contributors?: PackageJson.Person[];
+
+	/**
+	A list of people who maintain the package.
+	*/
+	maintainers?: PackageJson.Person[];
+
+	/**
+	The files included in the package.
+	*/
+	files?: string[];
+
+	/**
+	The module ID that is the primary entry point to the program.
+	*/
+	main?: string;
+
+	/**
+	An ECMAScript module ID that is the primary entry point to the program.
+	*/
+	module?: string;
+
+	/**
+	A module ID with untranspiled code that is the primary entry point to the program.
+	*/
+	esnext?:
+	| string
+	| {
+		main?: string;
+		browser?: string;
+		[moduleName: string]: string | undefined;
+	};
+
+	/**
+	A hint to JavaScript bundlers or component tools when packaging modules for client side use.
+	 */
+	browser?:
+	| string
+	| {
+		[moduleName: string]: string | false;
+	};
```

Comment:
`module`, `esnext`, and `browser` are also not official fields and should be in the intersection.